### PR TITLE
CORE-11865: Unify virtual node operation status

### PR DESCRIPTION
**Pull request by josephzunigadaly**
* Delete `VirtualNodeStatusCacheService`
* Change `findVirtualNodeOperationByRequestId` to only return the most recent status
* Change calling code to expect a single result from `findVirtualNodeOperationByRequestId`
* Change tests to expect a single result from `findVirtualNodeOperationByRequestId`
* Add `CREATE` to the get status response code paths
* Make `publishStatusMessage` write to `virtualNodeRepository`
* Update reading of create vnode status

API PR: https://github.com/corda/corda-api/pull/1552

### CODE REVIEW

**Comment by josephzunigadaly** [comment]
build e2e

**Comment by josephzunigadaly** [comment]
build e2e

**Comment by josephzunigadaly** [comment]
build e2e

**Comment by josephzunigadaly** [comment]
build e2e

File: components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/asyncoperation/handlers/AbstractVirtualNodeOperationHandler.kt [hunk 1]
```diff
@@ -1,46 +1,84 @@
...
 
 internal abstract class AbstractVirtualNodeOperationHandler(
-    private val statusPublisher: Publisher,
-    private val logger: Logger
+    private val entityManagerFactory: EntityManagerFactory,
+    private val virtualNodeRepository: VirtualNodeRepository = VirtualNodeRepositoryImpl()
 ) {
-    protected fun publishStartProcessingStatus(requestId: String) {
-        publishStatusMessage(requestId, getAvroStatusObject(requestId, VirtualNodeOperationStateDto.IN_PROGRESS))
+    protected fun publishStartProcessingStatus(requestId: String, virtualNodeOperationType: VirtualNodeOperationType) {
```
Comment (vkolomeyko) [inline components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/asyncoperation/handlers/AbstractVirtualNodeOperationHandler.kt]: IMO using term "publish" is not quite right here, as we are not really publishing anything in a sense of broadcasting. Therefore, perhaps better:
```suggestion
    protected fun recordStartProcessingStatus(requestId: String, virtualNodeOperationType: VirtualNodeOperationType) {
```
and rename all the "publish" methods below.

Comment (josephzunigadaly) [reply components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/asyncoperation/handlers/AbstractVirtualNodeOperationHandler.kt]: I have renamed these methods

**Review by vkolomeyko — DISMISSED** [review]
LGTM

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**